src/cppsc2_interface/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+# Single target interface library for bots to link against
+add_library(cppsc2_interface INTERFACE)
+
+target_link_libraries(cppsc2_interface
```

Comment:
Could you please briefly describe what's the idea behind creating this interface as a separate library?
From what I've seen in the modern c++ libs approach with targets (e.g. https://git.mts.ai/ai/voiceplatform/audio-transcoder/-/blob/develop/cmake/FindgRPC.cmake#L205) .

Reply:
Take the current CMake configuration for [BlankBot's src dir](https://github.com/cpp-sc2/blank-bot/blob/main/src/CMakeLists.txt):
```
# Create the executable.
add_executable(BlankBot ${BOT_SOURCES})
target_link_libraries(BlankBot
    sc2api sc2lib sc2utils sc2protocol civetweb libprotobuf
)
```

Some of these targets are cleaned up already (in particular protobuf through sc2protocol, and then alongside civetweb into sc2api, which would then consolidate the configuration to:
```
# Create the executable.
add_executable(BlankBot ${BOT_SOURCES})
target_link_libraries(BlankBot
    sc2api sc2lib sc2utils
)
```

I think this is fine, but does not accurately convey the intent of the `cpp-sc2` project usage. While `sc2api`, `sc2lib` and `sc2utils` are three separate compilation units, there are few independent use cases (one being pure bot library code, such as `sc2lib` which only needs to link to `sc2api`). But for an individual who just wants to resolve all dependencies required for a successful bot build from the boilerplate template, `cppsc2_interface` conveys the intent of these three libraries being the minimal targets required to use any of the API and library code.

Plus, there is no compilation overhead as there is no `cppsc2_interface` library artifact being produced as it is `INTERFACE` without sources (I've read about `INTERFACE` libraries generating other targets with custom commands, but that's outside of my knowledge and scope for this change).

I also have my own thoughts as to whether `sc2lib` even belongs in `cpp-sc2` as a dedicated library (I really think bot library code like this should be pulled into its own submodule, but the implementation released is so sparse and really exists to serve as an extension of the `examples`, providing users an idea of how to separate their own bot implementation from the project's).

A bit of a roundabout, but in summary:

- **Yes**, `cppsc2_interface` does not need to exist. Always linking against `sc2api`, `sc2utils`, and `sc2lib` when used is acceptable.
- `cppsc2_interface` conveys project intent, as to build a bot using `cpp-sc2` these three (but in reality only two are truly necessary, but the docs walkthrough requires `sc2lib` so we pull it in anyway) libraries are necessary.
- Especially in the case of pulling in `cpp-sc2` as a submodule (e.g. BlankBot) the library user shouldn't need to figure out what targets are necessary to use the project to its fullest. Providing a single interface target keeps things simple, for little build cost (I'm not in a position to quantify how many additional CPU cycles CMake will need to dedicate towards resolving the inheritance of one additional node in the tree)

Reply:
I see, thank you for the good description!
From my point of view there is no need in empty dir with single cmake file, but it is ok to keep it this way.

What do you think about renaming it into something like `cpp_sc2::cpp_sc2`? Just to use the same convention as other libraries e.g. fmt or gRPC.

Reply:
I think cpp_sc2 will be sufficient. `::` syntax is reserved for [imported or alias target names](https://cmake.org/cmake/help/latest/policy/CMP0028.html#policy:CMP0028).

The empty directory is unnecessary. I have no issues pushing the `INTERFACE` logic up into the `src` level build. Changes have been pushed.